adjust.go

```diff
@@ -154,73 +157,178 @@ func (f *File) adjustRowDimensions(sheet string, ws *xlsxWorksheet, row, offset
 	if newRow := lastRow.R + offset; lastRow.R >= row && newRow > 0 && newRow > TotalRows {
 		return ErrMaxRows
 	}
-	for i := 0; i < len(ws.SheetData.Row); i++ {
+	numOfRows := len(ws.SheetData.Row)
+	for i := 0; i < numOfRows; i++ {
 		r := &ws.SheetData.Row[i]
 		if newRow := r.R + offset; r.R >= row && newRow > 0 {
-			if err := f.adjustSingleRowDimensions(sheet, r, row, offset, false); err != nil {
-				return err
-			}
+			f.adjustSingleRowDimensions(sheet, r, offset)
+		}
+		if err := f.adjustSingleRowFormulas(sheet, r, row, offset, false); err != nil {
+			return err
 		}
 	}
 	return nil
 }
 
 // adjustSingleRowDimensions provides a function to adjust single row dimensions.
-func (f *File) adjustSingleRowDimensions(sheet string, r *xlsxRow, num, offset int, si bool) error {
+func (f *File) adjustSingleRowDimensions(sheet string, r *xlsxRow, offset int) {
 	r.R += offset
 	for i, col := range r.C {
 		colName, _, _ := SplitCellName(col.R)
 		r.C[i].R, _ = JoinCellName(colName, r.R)
+	}
+}
+
+// adjustSingleRowFormulas provides a function to adjust single row formulas.
+func (f *File) adjustSingleRowFormulas(sheet string, r *xlsxRow, num, offset int, si bool) error {
+	for _, col := range r.C {
 		if err := f.adjustFormula(sheet, col.F, rows, num, offset, si); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-// adjustFormula provides a function to adjust formula reference and shared
-// formula reference.
-func (f *File) adjustFormula(sheet string, formula *xlsxF, dir adjustDirection, num, offset int, si bool) error {
-	if formula == nil {
-		return nil
+// adjustCellRef provides a function to adjust cell reference.
+func (f *File) adjustRef(ref string, dir adjustDirection, num, offset int) (string, error) {
+	coordinates, err := rangeRefToCoordinates(ref)
+	if err != nil {
+		return ref, err
 	}
-	adjustRef := func(ref string) (string, error) {
-		coordinates, err := rangeRefToCoordinates(ref)
-		if err != nil {
-			return ref, err
-		}
-		if dir == columns {
+	if dir == columns {
+		if coordinates[0] >= num {
 			coordinates[0] += offset
+		}
+		if coordinates[2] >= num {
 			coordinates[2] += offset
-		} else {
+		}
+	} else {
+		if coordinates[1] >= num {
 			coordinates[1] += offset
+		}
+		if coordinates[3] >= num {
 			coordinates[3] += offset
 		}
-		return f.coordinatesToRangeRef(coordinates)
 	}
+	return f.coordinatesToRangeRef(coordinates)
+}
+
+// adjustRefToEntireRows provides a function to adjust reference to entire rows.
+func (f *File) adjustRefToEntireRows(ref string, num, offset int) string {
```

Comment:
In order to categorize the internally adjusted series functions, I suggest all the formula adjusting related functions name using adjustFormula prefix, so rename the following functions:

adjustRef -> adjustCellRef
adjustFormulaRef -> adjustFormulaCell
adjustRefToEntireRows -> adjustFormulaRows
adjustRefToEntireCols -> adjustFormulaCols